metals/src/main/scala/scala/meta/internal/metals/ScalafixProvider.scala

```diff
@@ -0,0 +1,184 @@
+package scala.meta.internal.metals
+
+import java.net.URLClassLoader
+import java.nio.file.Path
+import java.util.Collections
+import java.util.{List => JList}
+
+import scala.collection.concurrent.TrieMap
+import scala.util.Try
+
+import scala.meta._
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.io.AbsolutePath
+
+import org.eclipse.lsp4j.MessageType
+import org.eclipse.{lsp4j => l}
+import scalafix.interfaces.Scalafix
+import scalafix.interfaces.ScalafixArguments
+
+case class ScalafixProvider(
+    buildTargets: BuildTargets,
+    buffers: Buffers,
+    scalafixConfigPath: RelativePath,
+    workspace: AbsolutePath,
+    embedded: Embedded,
+    statusBar: StatusBar,
+    icons: Icons,
+    languageClient: MetalsLanguageClient
+) {
+  import ScalafixProvider._
+  private val scalafixCache = TrieMap.empty[ScalaBinaryVersion, Scalafix]
+  private val organizeImportRuleCache =
+    TrieMap.empty[ScalaBinaryVersion, List[Path]]
+
+  def organizeImports(file: AbsolutePath): List[l.TextEdit] = {
+    val fileInput = file.toInput
+    val unsavedFile = file.toInputFromBuffers(buffers)
+    if (isUnsaved(unsavedFile.text, fileInput.text)) {
+      scribe.info(s"Organize imports requires saving the file first")
+      languageClient.showMessage(
+        MessageType.Warning,
+        s"Save ${file.toNIO.getFileName} before organizing imports"
+      )
+      Nil
+    } else {
+      if (Seq("scala", "sbt").contains(file.`extension`)) {
+        val scalafixConfPath = workspace.resolve(scalafixConfigPath)
+        val scalafixConf =
+          if (scalafixConfPath.isFile) Some(scalafixConfPath.toNIO)
+          else None
+        val scalafixEvaluation = for {
+          (scalaVersion, scalaBinaryVersion, classPath) <-
+            scalaVersionAndClasspath(file)
+          _ <-
+            if (!ScalaVersions.isScala3Version(scalaVersion)) Some(())
```

Comment:
Could we move it to `scalaVersionAndClasspath` there would be no need for `Some(())`. We can return None there if the scalaversion is wrong.